[jdbc] ModifiablePersistenceService interface is not working for Postgre implementation

## Expected Behavior
ModifiablePersistenceService interface has been implemented for jdbc persistence at version 3.3.0.
I am developping a binding that needs this feature for being able to perform a synchronization of previously recorded data.
When calling the specific call of this interface designed to specify a date and time when to store data, provided value should be added to the database.

## Current Behavior
Sadly it happened it was buggy with Postgre implementation (at least... I did not check others).
Calling the store function from the ModifiablePersistenceService interface generates the following SQL exception:
```
2022-07-13 08:50:12.936 [DEBUG] [nhab.persistence.jdbc.db.JdbcBaseDAO] - JDBC::doStoreItemValue sql=INSERT INTO item0008 (TIME, VALUE) VALUES( ?, CAST( ? as #dbType#) ) timestamp=2022-03-07 00:00:00.0 value='46.3'
2022-07-13 08:50:12.939 [ERROR] [org.knowm.yank.Yank                 ] - Error in SQL query!!!
java.sql.SQLException: ERROR: syntax error at or near "#"
  Position: 60 Query: INSERT INTO item0008 (TIME, VALUE) VALUES( ?, CAST( ? as #dbType#) ) Parameters: [2022-03-07 00:00:00.0, 46.3, 46.3]
	at org.apache.commons.dbutils.AbstractQueryRunner.rethrow(AbstractQueryRunner.java:392) ~[?:?]
	at org.apache.commons.dbutils.QueryRunner.update(QueryRunner.java:491) ~[?:?]
	at org.apache.commons.dbutils.QueryRunner.update(QueryRunner.java:457) ~[?:?]
	at org.knowm.yank.Yank.execute(Yank.java:194) ~[?:?]
	at org.knowm.yank.Yank.execute(Yank.java:177) ~[?:?]
	at org.openhab.persistence.jdbc.db.JdbcBaseDAO.doStoreItemValue(JdbcBaseDAO.java:348) ~[?:?]
	at org.openhab.persistence.jdbc.internal.JdbcMapper.storeItemValue(JdbcMapper.java:165) ~[?:?]
	at org.openhab.persistence.jdbc.internal.JdbcPersistenceService.internalStore(JdbcPersistenceService.java:156) ~[?:?]
```

## Possible Solution
I entered in debugging mode with jdbc persistence addon, I found that the implementation of the store method from ModifiablePersistenceService was implemented only in the base class and there where no override for it in the Postgre class.

Actually the initialized SQL request for inserting is too different from the Postgre implementation and it expects a cast to the value type. So it declare inside the request a **#dbType#** tag which is never replaced by the base implementation and generates obviously an SQL error.

To quickly fix and continue working I implemented a possible override which has been tested and is working well.
Here it is as a reference for your coming fix (from JdbcPostgresqlDAO.java):
```
@Override
public void doStoreItemValue(Item item, State itemState, ItemVO vo, ZonedDateTime date) {
    ItemVO storedVO = storeItemValueProvider(item, itemState, vo);
    String sql = StringUtilsExt.replaceArrayMerge(sqlInsertItemValue,
            new String[] { "#tableName#", "#dbType#", "#tablePrimaryValue#" },
            new String[] { storedVO.getTableName(), storedVO.getDbType(), "?" });
    java.sql.Timestamp timestamp = new java.sql.Timestamp(date.toInstant().toEpochMilli());
    Object[] params = new Object[] { timestamp, storedVO.getValue() };
    logger.debug("JDBC::doStoreItemValue sql={} timestamp={} value='{}'", sql, timestamp, storedVO.getValue());
    Yank.execute(sql, params);
}
```

## Context
I am developping a binding that needs ModifiablePersistenceService interface for being able to perform a synchronization of previously recorded data.

## Your Environment
OpenHAB 3.3.0 on Linux Ubuntu

--- a/bundles/org.openhab.persistence.jdbc/src/main/java/org/openhab/persistence/jdbc/internal/db/JdbcPostgresqlDAO.java
+++ b/bundles/org.openhab.persistence.jdbc/src/main/java/org/openhab/persistence/jdbc/internal/db/JdbcPostgresqlDAO.java
@@ -13,6 +13,7 @@
 package org.openhab.persistence.jdbc.internal.db;
 
 import java.time.ZoneId;
+import java.time.ZonedDateTime;
 import java.util.List;
 
 import org.eclipse.jdt.annotation.NonNullByDefault;
@@ -160,6 +161,22 @@ public class JdbcPostgresqlDAO extends JdbcBaseDAO {
                 new String[] { storedVO.getTableName(), storedVO.getDbType(), sqlTypes.get("tablePrimaryValue") });
         Object[] params = { storedVO.getValue() };
         logger.debug("JDBC::doStoreItemValue sql={} value='{}'", sql, storedVO.getValue());
+        try {
+            Yank.execute(sql, params);
+        } catch (YankSQLException e) {
+            throw new JdbcSQLException(e);
+        }
+    }
+
+    @Override
+    public void doStoreItemValue(Item item, State itemState, ItemVO vo, ZonedDateTime date) throws JdbcSQLException {
+        ItemVO storedVO = storeItemValueProvider(item, itemState, vo);
+        String sql = StringUtilsExt.replaceArrayMerge(sqlInsertItemValue,
+                new String[] { "#tableName#", "#dbType#", "#tablePrimaryValue#" },
+                new String[] { storedVO.getTableName(), storedVO.getDbType(), "?" });
+        java.sql.Timestamp timestamp = new java.sql.Timestamp(date.toInstant().toEpochMilli());
+        Object[] params = { timestamp, storedVO.getValue() };
+        logger.debug("JDBC::doStoreItemValue sql={} timestamp={} value='{}'", sql, timestamp, storedVO.getValue());
         try {
             Yank.execute(sql, params);
         } catch (YankSQLException e) {

